error.go

```diff
@@ -79,21 +80,24 @@ func (d *ErrorDetail) Type() string {
 	//
 	// If we ever want to support remote registries, we can add an explicit
 	// `TypeURL` method.
-	return typeNameFromURL(d.pb.GetTypeUrl())
+	return typeNameFromURL(d.pbAny.GetTypeUrl())
 }
 
 // Bytes returns a copy of the Protobuf-serialized detail.
 func (d *ErrorDetail) Bytes() []byte {
-	out := make([]byte, len(d.pb.GetValue()))
-	copy(out, d.pb.GetValue())
+	out := make([]byte, len(d.pbAny.GetValue()))
+	copy(out, d.pbAny.GetValue())
 	return out
 }
 
 // Value uses the Protobuf runtime's package-global registry to unmarshal the
 // Detail into a strongly-typed message. Typically, clients use Go type
 // assertions to cast from the proto.Message interface to concrete types.
 func (d *ErrorDetail) Value() (proto.Message, error) {
-	return d.pb.UnmarshalNew()
+	if d.pbInner != nil {
+		return d.pbInner, nil
+	}
+	return d.pbAny.UnmarshalNew()
```

Comment:
Previously, the returned messages were safe to mutate b/c they were pointers to freshly-allocated structs. It feels like we may want to preserve that property and not expose `pbInner`?

Reply:
Yeah, that's a good point.

Reply:
Done. I've changed it to use `proto.Clone` because it should be a little more efficient than `*anypb.Any.UnmarshalNew`. But for serializing, we can still directly refer to `pbInner` since that won't expose it to potential mutations.